### DISCUSSION TITLE
Github.com navigation is slow with JavaScript enabled.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#21023] [Other Feature Feedback, Questions, & Ideas]
I just experienced how fast navigating around Github _used to be_ by simply turning off JavaScript. This is exciting to me! I'm sick to death of slow SPA navigation on websites.

I'm going to leave it disabled now in my main browser profile.

Does anyone know a Chrome extension to let me disable the SPA style navigation without completely disabling JavaScript?

**Comment by jfullstackdev** [comment]
I have never encountered GitHub slowing down. Maybe it's in your local setup (low memory, firewall restriction, extensions installed, unsupported browser ) or network connection. It's actually very fast even when just using a mobile hotspot. I'm using Chrome. Even GitHub Codespaces is very fast, given that it's a virtual machine we are accessing, but still will do the job without lagging.

**Comment by MrSarno** [comment]
I'd like to try to help, but it's not immediately clear what it is you are looking for.

GitHub is not a Single Page Application. Could you describe the specific behaviour you would like to avoid? It may be possible to achieve such through a userscript and an extension such as Tampermonkey. To create one, we'd need to better understand your request.

**Reply by ghost** [reply]
Hello! When I click around on Github with JS I can see the blue progress bar at the top of the page and many times, it just takes too long to get to the next page. Yesterday in my office, I noticed that turning off JS made navigation work faster. Today at home, Github appears to be navigating very quickly even with JS enabled but I have experienced slow navigation at home.

Another related annoyance is that with JS/HTML5 history push navigation I have definitely experienced loss of history before when I navigate too fast using back/foward buttons.

The next time I experience this issue, I can stop and investigate network and chrome extension issues before deploying the nuclear option. I doubt it's a network issue though, since JS-less navigation worked faster for me.

Thanks

**Reply by bert-w** [reply]
Same issue here. GitHub is inherently slow while navigating. I am using Firefox atm. This happened somewhere over the past few months. I waited it out for a while but it still remains as is.

GitHub might not be a Single Page Application but it behaves like one when navigation is handled through modern browser history APIs used by React.

**Reply by jfullstackdev** [reply]
would you think it should be on GitHub side improvement or the browser ? because I think, that's about the browser, because GitHub is working fine in Chrome

other browsers may not follow the standards, some of it or mostly all of it, that's why Internet Explorer came to an end

even in the online Counter-Strike, Chrome is definitely the number one supported, others would not simply work or too slow

**Reply by bert-w** [reply]
I don't think its a good idea to offer an experience tailored to Chrome only. Chrome browser market share is too big to my liking and it is up to developers to maintain the value of other browsers as well. Not everything has to be in Google's hands.

**Reply by jfullstackdev** [reply]
it's because they are not following the standards, that's why, it's the consequence, not because other developers like to support Chrome but they are not willing to make a site or game just to realize it's not working in other browsers, anyways that's just my opinion

**Reply by bert-w** [reply]
Different browsers have different Javascript engines which obviously changes things. I dont know if its about missing standards in this case since browser support and features has been very good for the past 5-10 years (compared to before that). Also my gut feeling says that GitHub users use more different browsers than the general user (which uses mainly Chrome).

**Reply by jfullstackdev** [reply]
> browser support and features has been very good for the past 5-10 years (compared to before that)

yup, and it's always changing as we all know, the same thing happened to those who failed to be updated all the time, maybe it's intentional or maybe they cannot catch up anymore
